src/sql-parser/src/parser.rs

```diff
@@ -2849,6 +2822,60 @@ impl<'a> Parser<'a> {
         }))
     }
 
+    fn parse_alter_index(&mut self) -> Result<Statement<Raw>, ParserError> {
+        let if_exists = self.parse_if_exists()?;
+        let name = self.parse_object_name()?;
+
+        match self.expect_one_of_keywords(&[RESET, SET, RENAME]) {
+            Ok(RESET) => {
+                self.expect_token(&Token::LParen)?;
+                let reset_options = self.parse_comma_separated(Parser::parse_identifier)?;
+                self.expect_token(&Token::RParen)?;
+
+                return Ok(Statement::AlterIndex(AlterIndexStatement {
```

Comment:
because all of these match conditions represent "terminal" states, you don't need to specify `return` on any of them, and could, in fact, wrap the entire `match` in an `Ok` and just return the `Statement::AlterIndex`es implicitly